inc/Engine/CriticalPath/APIClient.php

```diff
@@ -0,0 +1,234 @@
+<?php
+namespace WP_Rocket\Engine\CriticalPath;
+
+use WP_Error;
+
+/**
+ * Class APIClient
+ *
+ * @package WP_Rocket\Engine\CriticalPath
+ */
+class APIClient {
+
+	/**
+	 * Constant url for Critical Path API job.
+	 */
+	const API_URL = 'https://cpcss.wp-rocket.me/api/job/';
+
+	/**
+	 * Sends a generation request to the Critical Path API
+	 *
+	 * @since 3.6
+	 *
+	 * @param string $post_url The post URL.
+	 * @return array
+	 */
+	public function send_generation_request( $post_url ) {
+		$response = wp_remote_post(
+			self::API_URL,
+			[
+				// This filter is documented in inc/Engine/CriticalPath/CriticalCSSGeneration.php.
+				'body' => apply_filters(
+					'rocket_cpcss_job_request',
+					[
+						'url' => $post_url,
+					]
+				),
+			]
+		);
+
+		return $this->prepare_response( $response, $post_url );
+	}
+
+	/**
+	 * Prepare the response to be returned.
+	 *
+	 * @param array|WP_Error $response The response or WP_Error on failure.
+	 * @param string         $post_url Url for the post to be checked.
+	 * @return array|WP_Error
+	 * @since 3.6
+	 */
+	private function prepare_response( $response, $post_url ) {
+		$response_data        = $this->get_response_data( $response );
+		$response_status_code = $this->get_response_status( $response );
+		$succeeded            = $this->get_response_success( $response_status_code, $response_data );
+
+		if ( $succeeded ) {
+			return $response_data;
+		}else {
```

Comment:
You don't need the `else` here, since if the first condition is successful, you `return`. It will stop the method execution.

It will make the code easier to read 👍